src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,35 +38,42 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${todo.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
               type="button"
               className="delete"
               data-cy="modal-close"
+              onClick={() => {
+                onClose();
+                setUser(null);
```

Comment:
Consider moving the 'setUser(null)' to a cleanup function in the useEffect instead of resetting it in the onClick event.